packages/google_sign_in/google_sign_in/example/lib/main.dart

```diff
@@ -38,20 +42,36 @@ class SignInDemo extends StatefulWidget {
 
 class SignInDemoState extends State<SignInDemo> {
   GoogleSignInAccount? _currentUser;
+  bool _isAuthorized = false; // has granted permissions?
   String _contactText = '';
 
   @override
   void initState() {
     super.initState();
-    _googleSignIn.onCurrentUserChanged.listen((GoogleSignInAccount? account) {
+
+    _googleSignIn.onCurrentUserChanged
```

Comment:
This part seems not to work on web currently. 

web example uses `GoogleSignInUserData` and `_plugin.userDataEvents?.listen(...` but here we run into the missing idToken issue for handling AccessScopes

Reply:
tested this version

```
GoogleSignInUserData? _userData;

_plugin.userDataEvents?.listen((GoogleSignInUserData? userData) async {
  // Check if the account can access scopes...
  bool isAuthorized = false;
  if (userData != null) {
    if (userData.idToken != "") {
      isAuthorized =
          await _plugin.canAccessScopes(userData.idToken, scopes);
    }
  }
  setState(() {
    _userData = userData;
    _isAuthorized = isAuthorized;
  });
});
```

that seems to work.